File: suites/osprey-compute-kernels/reference/cases/josephus/josephus.hs
-- Josephus problem — survivor index for n people, step k=7, via the modular recurrence.
module Main (main) where

step :: Int -> Int -> Int
step acc i = (acc + 7) `mod` i

main :: IO ()
main = print (foldl step 0 [2 .. 10000000])

File: suites/osprey-compute-kernels/reference/cases/josephus/josephus.rs
// Josephus problem — survivor index for n people, step k=7, via the modular recurrence.
fn main() {
    let mut acc: i64 = 0;
    let mut i: i64 = 2;
    while i < 10000001 {
        acc = (acc + 7) % i;
        i += 1;
    }
    println!("{}", acc);
}
